http/server.ts

```diff
@@ -271,6 +340,10 @@ export class ServerRequest {
     }
 
     await this.w.flush();
+
+    if (!this._shouldReuseConnection()) {
+      this.conn.close();
```

Comment:
@ry you said that [`closeWriteAndWait`](https://github.com/golang/go/blob/9e277f7d554455e16ba3762541c53e9bfc1d8188/src/net/http/server.go#L1688) is what we need. I don't fully understand it's purpose. Could you follow up?